rclcpp/src/rclcpp/generic_publisher.cpp

```diff
@@ -23,6 +23,10 @@ namespace rclcpp
 
 void GenericPublisher::publish(const rclcpp::SerializedMessage & message)
 {
+  TRACETOOLS_TRACEPOINT(
+    rclcpp_publish,
+    nullptr,
+    static_cast<const void *>(&message.get_rcl_serialized_message()));
   auto return_code = rcl_publish_serialized_message(
     get_publisher_handle().get(), &message.get_rcl_serialized_message(), NULL);
```

Comment:
Shouldn't you also instrument the serialized version of take? I think it's in `SubscriptionBase::take_serialized()`.

Reply:
Exactly as you say. Fixed in https://github.com/ros2/rclcpp/pull/2448/commits/eae8938ff30c7031b181dc58a9243fd4e3e28337.